### DISCUSSION TITLE
Swift? MPS Graph?

### DISCUSSION THREAD

**Discussion by douglasdew** [ml-explore/mlx#39] [General]
I had recently begun an effort to port llama2.c to Swift, and was then going to improve that port to use MPS Graph. However, this new repo makes me wonder if I'm heading down a bad path.

Is Swift intended as a consumer of this repo's C++ apis?

What is the relationship of this repo to MPS and MPS Graph?

**Comment by awni** [comment]
1. We will be adding a Swift API so if you're not in a rush, that could be a nice solution for you.
2. Swift is definitely intended as a possible front-end
3. No relation to MPS Graph. We also currently don't use any MPS kernels, but that may change in the future.

**Reply by dougdew64** [reply]
Thank you for the clarification @awni.

It's going to be fun to watch this repo evolve.

**Comment by alikhan-tech** [comment]
The relationship between Swift and this repository's C++ APIs can be for consumption, but it depends on how these APIs are designed and Swift's interoperability with C++. As for MPS Graph, it's a framework used for neural network tasks on Apple platforms. The repo might relate to optimizing operations for MPS Graph within its C++ codebase. Whether Swift directly uses these C++ APIs varies based on project needs. To align your porting efforts, check the repo's documentation or engage with contributors for clarity on its intended use with Swift and MPS Graph.
